web-ui/src/components/reviews/periods/ReviewPeriods.jsx

```diff
@@ -1,162 +1,204 @@
+  ADD_REVIEW_PERIOD
+} from '../../../context/actions';
 import {
   selectCsrfToken,
   selectUserProfile,
   selectCurrentUserId,
   selectReviewPeriod,
-  selectReviewPeriods,
-} from "../../../context/selectors";
+  selectReviewPeriods
+} from '../../../context/selectors';
 
 const propTypes = {
   message: PropTypes.string,
-  onSelect: PropTypes.func,
+  onSelect: PropTypes.func
 };
-const displayName = "ReviewPeriods";
+const displayName = 'ReviewPeriods';
+const feedbackTemplateUrl = '/services/feedback/templates';
 
 const PREFIX = displayName;
 const classes = {
   actionButtons: `${PREFIX}-actionButtons`,
   headerContainer: `${PREFIX}-headerContainer`,
-  periodModal: `${PREFIX}-periodModal`,
+  periodModal: `${PREFIX}-periodModal`
 };
 
 const modalStyles = {
-  position: "absolute",
-  minWidth: "400px",
-  maxWidth: "600px",
-  backgroundColor: "background.paper",
-  top: "50%",
-  left: "50%",
-  padding: "1rem",
-  transform: "translate(-50%, -50%)",
-  border: "2px solid #fff",
+  position: 'absolute',
+  minWidth: '400px',
+  maxWidth: '600px',
+  backgroundColor: 'background.paper',
+  top: '50%',
+  left: '50%',
+  padding: '1rem',
+  transform: 'translate(-50%, -50%)',
+  border: '2px solid #fff',
+
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '1rem'
 };
 
 const modalActionStyles = {
-  marginTop: "1rem",
-  width: "calc(100% - 1rem)",
-  display: "flex",
-  flexDirection: "row",
-  justifyContent: "flex-end",
+  width: 'calc(100% - 1rem)',
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'flex-end'
 };
 
 const Root = styled('div')(({theme}) => ({
   [`& .${classes.actionButtons}`]: {
-    margin: "0.5em 0 0 1em",
-    ['@media (max-width:820px)']: { // eslint-disable-line no-useless-computed-key
-      padding: "0",
-    },
+    margin: '0.5em 0 0 1em',
+    ['@media (max-width:820px)']: {
+      // eslint-disable-line no-useless-computed-key
+      padding: '0'
+    }
   },
   [`& .${classes.headerContainer}`]: {
-    display: "flex",
-    flexDirection: "row",
-    justifyContent: "space-between",
-    alignItems: "center",
-    margin: "0 0 1em 0",
-  },
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    margin: '0 0 1em 0'
+  }
 }));
 
-const ReviewPeriods = ({ onPeriodSelected, mode }) => {
-  const { state, dispatch } = useContext(AppContext);
-  const userProfile = selectUserProfile(state);
-  const currentUserId = selectCurrentUserId(state);
-  const csrf = selectCsrfToken(state);
-  const periods = selectReviewPeriods(state);
-  const [selfReviews, setSelfReviews] = useState(null);
-  const [periodToAdd, setPeriodToAdd] = useState({ name: "", open: true });
-  const isAdmin = userProfile?.role?.includes("ADMIN");
-  const [open, setOpen] = useState(false)
+const ReviewPeriods = ({onPeriodSelected, mode}) => {
+  const {state, dispatch} = useContext(AppContext);
+
+  const [canSave, setCanSave] = useState(false);
```

Comment:
This is a new bit of state that is used to conditionally disable the Save button.

Reply:
Nice.